packages/react/src/useEnqueue.ts

```diff
@@ -0,0 +1,106 @@
+import type { EnqueueBackgroundActionOptions, GadgetConnection } from "@gadgetinc/api-client-core";
+import {
+  BackgroundActionHandle,
+  disambiguateActionVariables,
+  disambiguateBulkActionVariables,
+  enqueueActionOperation,
+  get,
+  graphqlizeBackgroundOptions,
+} from "@gadgetinc/api-client-core";
+import type { AnyActionFunction } from "@gadgetinc/api-client-core/src/GadgetFunctions.js";
+import { useCallback, useContext, useMemo } from "react";
+import type { UseMutationState } from "urql";
+import { GadgetUrqlClientContext, useConnection } from "./GadgetProvider.js";
+import { useGadgetMutation } from "./useGadgetMutation.js";
+import type { EnqueueHookResult, EnqueueHookState } from "./utils.js";
+import { ErrorWrapper, noProviderErrorMessage } from "./utils.js";
+
+/**
+ * React hook to enqueue a Gadget action in the background. `useEnqueue` must be passed an action function from an instance of your generated API client library, like `useEnqueue(api.user.create)` or `useEnqueue(api.someGlobalAction)`. `useEnqueue` doesn't actually submit the background action when invoked, but instead returns a function for enqueuing the action in response to an event.
+ *
+ * @param action a model action or global action in your application's client, like `api.user.create` or api.someGlobalAction`
+ * @param options action options, like selecting the fields in the result
+ *
+ * @example
+ * export function CreateUserButton(props: { name: string; email: string }) {
+ *   const [{error, fetching, handle}, runEnqueue] = useEnqueue(api.user.create));
+ *
+ *   const onClick = () => runEnqueue(
```

Comment:
super nit: I read this as use`enqueue`, rather than use`runEnqueue`

```suggestion
 *   const [{error, fetching, handle}, enqueue] = useEnqueue(api.user.create));
 *
 *   const onClick = () => enqueue(
```